calm-visualizer/src/components/graph/Graph.tsx

```diff
@@ -2,94 +2,123 @@ import { useEffect } from "react";
-        anchor: "Continuous",
-        connector: {
-            type:"Straight",
-            options: {
-                "stub": 25
-            }
-        },
-        endpoint: {
-            type: "Blank",
-            options: {}
-        },
-        overlays: [
-            { type:"Arrow", options:{location:1}}
-        ]
-    });
+                const r  = relationship["relationship-type"]["connects"];
+                instance.connect({
+                    source: document.getElementById(r.source.node)!,
+                    target: document.getElementById(r.destination.node)!,
+                    anchor: "Continuous",
+                    connector: {
+                        type:"Straight",
+                        options: {
+                            "stub": 25
+                        }
+                    },
+                    endpoint: {
+                        type: "Blank",
+                        options: {}
+                    },
+                    overlays: [
+                        { type:"Arrow", options:{location:1}}
+                    ]
+                });
 }
 
 function createInteractsRelationship(instance: BrowserJsPlumbInstance, relationship: CALMInteractsRelationship) {
-    const r = relationship["relationship-type"]["interacts"];
-    r.nodes.forEach(node => {
-        instance.connect({
-            source: document.getElementById(r.actor)!,
-            target: document.getElementById(node)!,
-            anchor: "Continuous",
-            connector: {
-                type:"Straight",
-                options: {
-                    "stub": 25
-                }
-            },
-            endpoint: {
-                type: "Blank",
-                options: {
-                }
-            },
-            overlays: [
-                { type:"Arrow", options:{location:1}}
-            ]
-        });
+        const r = relationship["relationship-type"]["interacts"];
+        r.nodes.forEach(node => {
+            instance.connect({
+                source: document.getElementById(r.actor)!,
+                target: document.getElementById(node)!,
+                anchor: "Continuous",
+                connector: {
+                    type:"Straight",
+                    options: {
+                        "stub": 25
+                    }
+                },
+                endpoint: {
+                    type: "Blank",
+                    options: {
+                    }
+                },
+                overlays:[ 
+                    { type:"Arrow", options:{location:1}}
+                ]
+            });
+        })
+}
+
+function getUniqueCALMContainers(relationships : CALMRelationship[]){
+    const deployedInContainers : string[] =  relationships.filter(relationship => 'deployed-in' in relationship["relationship-type"]).map(r=>r as CALMDeployedInRelationship).map(r=> r["relationship-type"]["deployed-in"].container);
```

Comment:
these lines are quite long, could we do each `.map(..)` on a new line?

Reply:
Resolved